packages/server/graphql/mutations/createPoll.ts

```diff
@@ -0,0 +1,129 @@
+import {GraphQLNonNull} from 'graphql'
+import {getUserId, isTeamMember} from '../../utils/authorization'
+import publish from '../../utils/publish'
+import CreatePollPayload from '../types/CreatePollPayload'
+import {SubscriptionChannel} from 'parabol-client/types/constEnums'
+
+import {GQLContext} from '../graphql'
+import CreatePollInput from '../types/CreatePollInput'
+import MeetingMemberId from 'parabol-client/shared/gqlIds/MeetingMemberId'
+import insertPoll from '../../postgres/queries/insertPoll'
+import insertPollOptions from '../../postgres/queries/insertPollOptions'
+import segmentIo from '../../utils/segmentIo'
+
+const MAX_POLL_OPTIONS = 4
+const MIN_POLL_OPTIONS = 2
+const MAX_TITLE_LENGTH = 100
+
+type PollOptionsInputVariables = {
+  title: string
+}
+
+type CreatePollInputVariables = {
+  newPoll: {
+    teamId: string
+    discussionId: string
+    title: string
+    threadSortOrder: number
+    options: PollOptionsInputVariables[]
+  }
+}
+
+const createPoll = {
+  type: GraphQLNonNull(CreatePollPayload),
+  args: {
+    newPoll: {
+      type: new GraphQLNonNull(CreatePollInput),
+      description: 'The new poll including title and poll options'
+    }
+  },
+  resolve: async (
+    _source,
+    {newPoll}: CreatePollInputVariables,
+    {authToken, dataLoader, socketId: mutatorId}: GQLContext
+  ) => {
+    const viewerId = getUserId(authToken)
+    const operationId = dataLoader.share()
+    const subOptions = {mutatorId, operationId}
+    const {teamId, discussionId, threadSortOrder, title, options} = newPoll
+
+    //AUTH
+    if (!isTeamMember(authToken, teamId)) {
+      return {error: {message: 'Not on team'}}
+    }
+
+    // VALIDATION
+    if (!title) {
```

Comment:
Should we also check if the `title` is a duplicate, i.e., can we create two identical polls?

Reply:
I'm personally don't think it should be an issue but I'm open to check that. @mattkrick @enriquesanchez any thoughts on that?

Reply:
that's a good consideration.
I feel like the `sortOrder` acts as the unique identifier in the UI. 
If they have multiple rounds of polls, removing 1 option each time, they might keep the same name? IMO that's kinda lame, but i could see folks doing it. I'm cool with leaving it out for now just to get it out & see what folks do with it

Reply:
can delete this conditional or change it to `title.length === 0` because title is guaranteed to be non-null from `CreatePollInput`

maybe we add a min length?